middleware/index.ts

```diff
@@ -264,3 +264,72 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
     // Function params
     return [event, context, logger];
 }
+
+function getUserFn() {
+    const functionPath = process.env.USER_FUNCTION_URI || '/workspace';
+    const pjsonPath = path.join(functionPath, 'package.json');
+    let main;
+    try {
+        main = require(pjsonPath).main;
+    } catch (e) {
+        console.log(e);
+        throw `Could not read package.json: ${e}`;
+    }
+    const mainPath = path.join(functionPath, main);
+    let mod;
+    try {
+        mod = require(mainPath);
+    } catch (e) {
+        console.log(e);
+        throw `Could not locate user function: ${e}`;
+    }
+    if (mod.__esModule && typeof mod.default === 'function') {
+        return mod.default;
+    }
+    return mod;
+}
+
+function createLogger(requestID: string): Logger {
```

Comment:
`requestID: string | undefined` ?

Reply:
It might actually be more idiomatic to do `requestID?: string`

Reply:
nice - shorter, too.